docs/content/python/guides/roles/getting-started.md

```diff
@@ -0,0 +1,273 @@
+to actors to grant them all the permissions the role has.
+
+We define resources in Polar using the `resource` rule. The `Org`
+resource represents an Organization in the GitClub example application.
+Let's walk through the resource definition for `Org`.
+
+```polar
+resource(_type: Org, "org", actions, roles) if
+```
+
+The rule head has 4 parameters:
+
+- `_type` is the Python class the resource definition is associated with.
+- `"org"` is the identifier for this resource type (this can be any string
+  you choose).
+- `actions` are set in the rule body. The `actions` list defines all the
+  actions that may be performed on the resource.
+- `roles` are set in the rule body. The `roles` dictionary defines all the
+  roles for this resource.
+
+In our rule body, we first define the list of available actions for this
+resource:
+
+```polar
+resource(_type: Org, "org", actions, roles) if
+    actions = ["read", "create_repo"] and
+    roles = {
+        ...
+    };
+```
+
+Now, we define our roles. Roles are defined in a dictionary that maps
+the role name to a role configuration.
+
+```polar
+resource(_type: Org, "org", actions, roles) if
+    actions = ["read", "create_repo"] and
+    roles = {
+        member: {
+            permissions: ["read"],
+        },
+        owner: {
+            permissions: ["read", "create_repo"],
+        }
+    };
+```
+
+This resource definition defines two roles:
+
+- _member_: Has the `read` permission.
+- _owner_: Has the `read` and `create_repo` permissions.
+
+Permissions are actions associated with a resource type. A permission can
+directly reference an action defined in the same resource. Later, we'll
+see how to add permissions defined on other resources to roles.
+
+{{% callout "resource(...) is just a rule" "blue" %}}
+
+The `resource` definition is just a regular Polar rule. That's why it
+has an `if` and `and` between variable assignments. `actions` and
+`roles` are unbound parameters, meaning they can be assigned inside of
+the rule body.
+
+We could have written this rule without a body, like:
+
+```polar
+resource( _type: Org, "org", ["read", "create_repo"],
+    {
+        member: {
+            permissions: ["read"],
+        },
+        owner: {
+            permissions: ["read", "create_repo"],
+        }
+    }
+);
+```
+
+But, we think the expanded form is clearer.
+
+{{% /callout %}}
+
+### Adding role_allows to our policy
+
+To allow access based on roles, we add the following `allow` rule
+
+```polar
+allow(actor, action, resource) if
+    role_allow(actor, action, resource);
+```
+
+Oso will now allow access to any resource that is allowed based on the
+role definitions.
+
+### Assigning roles to users
+
+Now we've configured roles and setup our policy. For users to have
+access, we must assign them to roles.
+
+We do this by writing an `actor_has_role_for_resource` rule.
+This is how you tell Oso what roles
+a user has.
+
+{{% callout "Managing roles with SQLAlchemy" "green" %}}
```

Comment:
Should maybe move this above "We do this by writing an `actor_has_role_for_resource` rule." so that SQLAlchemy users don't think they need to write that rule